pkg/kube/cluster-init.sh

```diff
@@ -7,59 +7,96 @@ K3S_VERSION=v1.26.3+k3s1
 KUBEVIRT_VERSION=v0.59.0
 LONGHORN_VERSION=v1.4.2
 CDI_VERSION=v1.56.0
+NODE_IP=""
 
 INSTALL_LOG=/var/lib/install.log
 CTRD_LOG=/var/lib/containerd.log
 LOG_SIZE=$((5*1024*1024))
 
 logmsg() {
-   local MSG
-   local TIME
-   MSG="$*"
-   TIME=$(date +"%F %T")
-   echo "$TIME : $MSG"  >> $INSTALL_LOG
+        local MSG
+        local TIME
+        MSG="$*"
+        TIME=$(date +"%F %T")
+        echo "$TIME : $MSG"  >> $INSTALL_LOG
 }
 
 setup_cgroup () {
-   echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
+        echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
 }
 
 
 check_network_connection () {
- while true; do
-
-   ret=$(curl -o /dev/null -w "%{http_code}" -s "https://get.k3s.io")
-   if [ "$ret" -eq 200 ]; then
-        logmsg "Network is ready."
-        break;
-   else
-        logmsg "Network is not yet ready"
-   fi
-
-   sleep 5
- done
+        while true; do
+                ret=$(curl -o /dev/null -w "%{http_code}" -s "https://get.k3s.io")
+                if [ "$ret" -eq 200 ]; then
+                        logmsg "Network is ready."
+                        break;
+                else
+                        logmsg "Network is not yet ready"
+                fi
+                sleep 5
+        done
 }
 
 wait_for_default_route() {
-  while read -r iface dest gw flags refcnt use metric mask mtu window irtt; do
-    if [ "$dest" = "00000000" ] && [ "$mask" = "00000000" ]; then
-      logmsg "Default route found"
-      return 0
-    fi
-    #Make yetus happy
-    logmsg "waiting for default route $iface $dest $gw $flags $refcnt $use $metric $mask $mtu $window $irtt"
-    sleep 1
-  done < /proc/net/route
-
-  return 1
+        while read -r iface dest gw flags refcnt use metric mask mtu window irtt; do
+                if [ "$dest" = "00000000" ] && [ "$mask" = "00000000" ]; then
+                        logmsg "Default route found"
+                        return 0
+                fi
+                logmsg "waiting for default route $iface $dest $gw $flags $refcnt $use $metric $mask $mtu $window $irtt"
+                sleep 1
+        done < /proc/net/route
+        return 1
+}
+
+# Get IP of the interface with the first default route
+# This will be then used as K3s node IP.
+get_default_intf_IP_prefix() {
+        logmsg "Trying to obtain Node IP..."
+        while [ -z "$NODE_IP" ]; do
+                # Find the default route interface
+                default_interface="$(ip route show default | head -n 1 | awk '/default/ {print $5}')"
```

Comment:
If a node has default routes out multiple interfaces (eth0, eth1, and wwan0 for example), will this always pick the same one each time the device is rebooted? Or could it vary from one reboot to the next?

Reply:
This would vary with each reboot. However, we run this only once when deploying multus and not on every reboot.
But note that this is just an initial implementation, eventually the port for kubernetes will be selected by the user.

Reply:
It would be good to add a comment that this wil go away when the user/controller can select the cluster network interface.

Reply:
Done